app/src/main/java/de/tum/in/tumcampusapp/component/ui/news/TopNewsViewModel.kt

```diff
@@ -16,27 +15,33 @@ import io.reactivex.schedulers.Schedulers
                        private val compositeDisposable: CompositeDisposable): ViewModel() {
 
-        /**
-         * Downloads the NewsAlert and stores it in the sharedPreferences
-         */
-        fun getNewsAlertFromService(context: Context): Boolean =
-        compositeDisposable.add(Observable.just(1)
-        .subscribeOn(Schedulers.computation())
-        .flatMap { remoteRepository.getNewsAlert() }.observeOn(Schedulers.io())
-        .doOnError { Utils.log(it.message) }
-        .subscribe({ t ->
-                        Utils.setSetting(context, Const.NEWS_ALERT_IMAGE, t.url)
-                        Utils.setSetting(context, Const.NEWS_ALERT_LINK, t.link)
+    /**
+     * Downloads the NewsAlert and stores it in the sharedPreferences
+     */
+    fun getNewsAlertFromService(context: Context): Boolean {
+        return compositeDisposable.add(
+                remoteRepository
+                        .getNewsAlert()
+                        .subscribeOn(Schedulers.computation())
+                        .observeOn(Schedulers.io())
+                        .doOnError { Utils.log(it) }
+                        .subscribe({ newsAlert ->
+                            // TODO: Store NewsAlerts in Room, not SharedPreferences!
+                            Utils.setSetting(context, Const.NEWS_ALERT_IMAGE, newsAlert.url)
+                            Utils.setSetting(context, Const.NEWS_ALERT_LINK, newsAlert.link)
 
-                        val oldShowUntil = Utils.getSetting(context, Const.NEWS_ALERT_SHOW_UNTIL, "")
-                        val oldImage = Utils.getSetting(context, Const.NEWS_ALERT_IMAGE, "");
+                            val oldShowUntil = Utils.getSetting(context, Const.NEWS_ALERT_SHOW_UNTIL, "")
+                            val oldImage = Utils.getSetting(context, Const.NEWS_ALERT_IMAGE, "");
 
-                        // there is a NewsAlert update if the image link or the date changed --> Card should be displayed again
-                        val update = !oldShowUntil.equals(t.displayUntil) || !oldImage.equals(t.url)
-                        if(update){
-                            Utils.setSetting(context, CardManager.SHOW_TOP_NEWS, true)
-                        }
-                        Utils.setSetting(context, Const.NEWS_ALERT_SHOW_UNTIL, t.displayUntil)
-                })
-        )}
+                            // There is a NewsAlert update if the image link or the date changed
+                            // --> Card should be displayed again
+                            val update = oldShowUntil != newsAlert.displayUntil || oldImage != newsAlert.url
+                            if (update) {
+                                Utils.setSetting(context, CardManager.SHOW_TOP_NEWS, true)
+                            }
+                            Utils.setSetting(context, Const.NEWS_ALERT_SHOW_UNTIL, newsAlert.displayUntil)
+                        }, { t -> Utils.log(t) })
```

Comment:
We already have a `.doOnError { Utils.log(it) }` call in here, we can probably remove that one then...

Reply:
You’re right – I forgot to remove that 🙈 

We want to go with the new one (`onError`), because it actually consumes the error. `doOnError` does not.

Reply:
So i guess search + replace `doOnError` -> `onError`

Reply:
Sorry, I should be clearer: There is no single `onError` method. It’s a `Consumer` that you pass to `subscribe`.

However, there are other ways to achieve the same thing: We could use `onErrorReturnItem` or `onErrorResumeNext` instead and return an empty `Observable` or `Flowable`. I’ll have to investigate if that makes any difference between those and my current implementation.

Reply:
@pfent In the meantime, please remove the approval for this branch again.

Reply:
So generally, we want to use an `onError` consumer, to properly consume everything...

I'm not a fan of `onErrorReturn` or `onErrorResumeNext`, since we probably should do proper error handling with dedicated UI elements that would be shown in the error subscriber...

Reply:
There are a few things here that make proper error handling difficult. Let’s talk in person tomorrow about how we can refactor this.